packages/next/server/web/spec-extension/request.ts

```diff
@@ -20,14 +20,15 @@ export class NextRequest extends Request {
   }
 
   constructor(input: Request | string, init: RequestInit = {}) {
+    const url = typeof input === 'string' ? input : input.url
+    validateURL(url)
```

Comment:
is this added by purpose? should we only valid in dev?

Reply:
We decided to keep the behavior for dev/prod to be consistent

Reply:
doesn't `NextURL` doing that already?